plugins/inputs/jolokia2/client.go

```diff
@@ -132,7 +132,8 @@ func (c *Client) read(requests []ReadRequest) ([]ReadResponse, error) {
 
 	req, err := http.NewRequest("POST", requestURL, bytes.NewBuffer(requestBody))
 	if err != nil {
-		return nil, fmt.Errorf("unable to create new request '%s': %s", requestURL, err)
+		//err is not contained in returned error - it may contain sensitive data (password) which should not be logged
+		return nil, fmt.Errorf("unable to create new request for: '%s'", c.URL)
```

Comment:
this one looks weird to me. not sure what to think of it.

Reply:
@ssoroka `requestURL` and/or `err` can contain password (from `c.config.Password`). Thanks to this change it won't be possible. If you have any other idea how to avoid it, I will follow your lead :)